modules/aruco/src/charuco.cpp

```diff
@@ -938,5 +938,62 @@ void drawDetectedDiamonds(InputOutputArray _image, InputArrayOfArrays _corners,
         }
     }
 }
+
+/**
+   @param board layout of ChArUco board.
+ * @param image charucoIds list of identifiers for each corner in charucoCorners.
+ * @return bool value, 1 (true) for detected corners form a line, 0 for non-linear.
+      solvePnP will fail if the corners are collinear (true).
+  * Check that the set of charuco markers in _charucoIds does not identify a straight line on
+    the charuco board.  Axis parallel, as well as diagonal and other straight lines detected.
+  */
+bool testCharucoCornersCollinear(const Ptr<CharucoBoard> &_board, InputArray _charucoIds){
+
+    unsigned int nCharucoCorners = (unsigned int)_charucoIds.getMat().total();
+
+    // only test if there are 3 or more corners
+    if (nCharucoCorners > 2){
```

Comment:
```
if (nCharucoCorners <= 2)
    return true;
```

Reply:
I switched the order of the test in this function -- which I think is what you're suggesting here.